packages/jest-haste-map/src/__tests__/haste_impl.js

```diff
@@ -23,4 +28,8 @@ module.exports = {
       .substr(filename.lastIndexOf(path.sep) + 1)
       .replace(/(\.(android|ios|native))?\.js$/, '');
   },
+
+  setCacheKey(key) {
+    cacheKey = key;
```

Comment:
The cache key shouldn't be setteable; we should follow the `transform` way, where only the getter exsists.

Reply:
This is a fake haste impl for testing. I added the method to test that when you change the cache key the cache file for the haste map is different.